ncdfchecker.py

```diff
@@ -189,16 +193,66 @@ def check_globals(product, constraints, skip=["short_name"], strict=False,
     return errcount, warncount
 
 
-def check_stepsize(data, stepsize):
+def get_month_midpoints(monthslist):
+    """Return a list of dates that correspond to the middle of the month."""
+    def midpoint(start, end): return start + (end - start) / 2
+
+    midmonths = [midpoint(monthslist[i], monthslist[i+1])
+                 for i in range(len(monthslist)-1)]
+
+    return midmonths
+
+
+def get_expected_monthly_stepsize(initdate, runlength):
```

Comment:
Initdate should be taken from the first entry in the item being validated - so is startdate rather than GloSea model based nomenclature of init date.

Reply:
As mentioned above the start date is taken from the meta-data "forecast_reference_time".